flashlight/fl/tensor/backend/jit/JitTensorBase.cpp

```diff
@@ -15,44 +15,56 @@
 
 namespace fl {
 
-JitTensorBase::JitTensorBase(Node* node) : node_(node) {
-  node_->incRefCount();
-}
+struct JitTensorBase::ShareData {
```

Comment:
doesn't `shared_ptr<Node>` handle this kind of stuff for you? curious why you're wrapping it twice

Reply:
Bleh -- is having `weak_ptr` actually that much worse if you're just dealing with `Node*` in the public API? It seems like the `ShareData` is just internal sugar to pass around refcount ops and users will still just want `Node*` to build their own passes

Also might be worth thinking about the level of technical-ness one could expect from someone implementing compiler passes 😄 

Probably should've put this thought on your prior PR.

Reply:
Had some discussions offline. Current plan (for a future refactor PR, after landing the remaining stack which will be orthogonal to this):

1. Use `std::shared_ptr<Node>` to replace manual refcounting.
2. Use `Node&` or `std::weak_ptr<Node>` inside `Use` (need to investigate more)
3. Use something other than refcount to determine whether to retain tensor result in Evaluator.